### PR TITLE
Fix INavigationChildrenProps's onPerPageClick

### PR DESCRIPTION
**Pull request by tommy-tran**
Changes:
- Change to onPerPageClick definition in Navigation and NavigationPerPage interfaces
- Added test for proper arguments being used in onPerPageClick
- Alert the value of onPerPageClick and onPageClick in component examples

### CODE REVIEW

**Comment by GermainBergeron** [comment]
It would be nice to add a test in `src/components/navigation/Navigation.spec.tsx` to validate that the parameters are correctly sent when the user change the "perPage"

File: src/components/navigation/Navigation.tsx [hunk 1]
```diff
@@ -21,7 +21,7 @@ export interface INavigationChildrenProps {
     onPageClick?: (pageNb: number) => void;
     perPageLabel?: string;
     perPageNumbers?: number[];
-    onPerPageClick?: () => void;
+    onPerPageClick?: (newPerPage: number, currentPerPage: number, currentPage: number) => void;
```
Comment (sroy3) [inline src/components/navigation/Navigation.tsx]: I think the parameters should be optional. The user might not need them and might send a function without these parameters.

Comment (GermainBergeron) [reply src/components/navigation/Navigation.tsx]: I think you're confused, the onPerPageClick is optional, but here we define what we send to the function. If the "user" don't want to use them they can simply omit them in the receiver

[Example](http://www.typescriptlang.org/play/#src=interface%20SomeProps%20%7B%0D%0A%20%20%20%20f%3F%3A%20(a%3A%20string%2C%20b%3A%20number)%20%3D%3E%20any%3B%0D%0A%7D%0D%0A%0D%0Aconst%20One%3A%20SomeProps%20%3D%20%7B%7D%3B%0D%0A%0D%0Aconst%20Two%3A%20SomeProps%20%3D%20%7B%0D%0A%20%20%20%20f%3A%20()%20%3D%3E%20null%2C%0D%0A%7D%3B%0D%0A%0D%0Aconst%20Three%3A%20SomeProps%20%3D%20%7B%0D%0A%20%20%20%20f%3A%20(a%3A%20string)%20%3D%3E%20a%2C%0D%0A%7D%0D%0A%0D%0Aconst%20Four%3A%20SomeProps%20%3D%20%7B%0D%0A%20%20%20%20f%3A%20(a%3A%20string%2C%20b%3A%20number)%20%3D%3E%20b%2C%0D%0A%7D)

File: src/components/navigation/tests/Navigation.spec.tsx [hunk 2]
```diff
@@ -82,5 +83,14 @@ describe(' navigation', () => {
...
             expect(navigation.find(NavigationPerPage).props().currentPerPage).toBe(expectedPerPage);
         });
+
+        it('should call onPerPageClick prop with the correct values when it is set', () => {
+            const onPerPageClick = jasmine.createSpy('mockOnPerPageClick');
+            const perPageNumbers: number[] = [2, 3, 4];
+            const currentPerPage: number = 3;
+            const newNavigationProps: INavigationProps = _.extend({}, basicNavigationProps, {currentPerPage, perPageNumbers, onPerPageClick});
+            navigation.setProps(newNavigationProps).update().find(NavigationPerPageSelect).last().simulate('click');
+            expect(onPerPageClick).toHaveBeenCalledWith(4, 3, undefined); // newPerPage, currentPerPage, currentPage (currentPerPage and currentPage are undefined unless passed as props)
```
Comment (GermainBergeron) [inline src/components/navigation/tests/Navigation.spec.tsx]: Test looks god, I'd change the expect a bit:

```typescript
const currentPerPage = perPageNumbers[1];
const expectedPerPage = perPageNumbers[perPageNumber.length - 1];
...
expect(onPerPageClick).toHaveBeenCalledWith(expectedPerPage, currentPerPage, undefined);
```
This way if you add/modify the perPageNumbers const the test will stay green

File: src/components/navigation/Navigation.tsx (hunk 1 above)
Comment (MichelML) [reply src/components/navigation/Navigation.tsx]: I thought the same as @sroy3 but it seems you're right according to your example

Comment (tommy-tran) [reply src/components/navigation/Navigation.tsx]: After looking at it again, I think that currentPerPage and currentPage are optional, and the only required one would be newPerPage. The other arguments are only set if Navigation receives currentPerPage and currentPage as props (in the current implementation).

**Pull request merged**